### PR TITLE
fix: handle YAML front matter block

### LINKED ISSUE #3321: SCSS: Preserve YAML front matter block
Prettier is smart enough to preserve YAML front matter blocks in Markdown files, but currently strips them from SCSS files, which breaks Jekyll sites such as those deployed via GitHub Pages.

Recent Jekyll versions provide native Sass support, but the Sass processing is only triggered if the file contains a valid YAML front matter block per [Jekyll docs](https://jekyllrb.com/docs/frontmatter/):

> Any file that contains a YAML front matter block will be processed by Jekyll as a special file. 

When Prettier removes the YAML header, site output appears unstyled, as the stylesheet is not parsed.

The same logic that allows Prettier to ignore YAML headers in Markdown files should be applied to SCSS.

### PR DESCRIPTION
**Pull request by evilebottnawi**
Fixes #3321

### CODE REVIEW

File: src/language-css/parser-postcss.js [hunk 1]
```diff
@@ -261,15 +274,29 @@ function parse(text, parsers, opts) {
     ? opts.parser === "scss"
     : IS_POSSIBLY_SCSS.test(text);
 
+  let frontMatter = null;
+
+  try {
+    frontMatter = grayMatter(text);
+  } catch (originalError) {
```
Comment (lydell) [inline src/language-css/parser-postcss.js]: Why is this caught error called `originalError`?

Comment (evilebottnawi) [reply src/language-css/parser-postcss.js]: @lydell other name?

Comment (lydell) [reply src/language-css/parser-postcss.js]: `error` I guess

**Comment by evilebottnawi** [comment]
@lydell done :+1:

File: src/language-css/parser-postcss.js [hunk 2]
```diff
@@ -261,15 +274,29 @@ function parse(text, parsers, opts) {
     ? opts.parser === "scss"
     : IS_POSSIBLY_SCSS.test(text);
 
+  let frontMatter = null;
+
+  try {
+    frontMatter = grayMatter(text);
+  } catch (error) {
+    throw error;
+  }
```
Comment (lydell) [inline src/language-css/parser-postcss.js]: Why try-catch if we immediately re-throw?

**Comment by evilebottnawi** [comment]
@lydell done :+1:

**Pull request merged**